src/dht_proxy_server.cpp

```diff
@@ -190,21 +220,28 @@ DhtProxyServer::listen(const std::shared_ptr<restbed::Session>& session) const
                     infoHash = InfoHash::get(hash);
                 }
                 s->yield(restbed::OK);
-                    // Handle client deconnection
-                    // NOTE: for now, there is no handler, so we test the session in a thread
-                    // will be the case in restbed 5.0
+                // Handle client deconnection
+                // NOTE: for now, there is no handler, so we test the session in a thread
+                // will be the case in restbed 5.0
                 SessionToHashToken listener;
                 listener.session = session;
                 listener.hash = infoHash;
-                listener.token = dht_->listen(infoHash, [s](std::shared_ptr<Value> value) {
+                // cache the session to avoid an incrementation of the shared_ptr's counter
+                // else, the session->close() will not close the socket.
+                auto cacheSession = std::weak_ptr<restbed::Session>(s);
+                listener.token = std::move(dht_->listen(infoHash, [cacheSession](std::shared_ptr<Value> value) {
```

Comment:
std::move is unecesary because the return value is already an r-value.

produces a warning:
warning: moving a temporary object prevents copy elision